plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,645 @@
+import static java.util.stream.Collectors.joining;
+
+public class IgniteJdbcClient
+        extends BaseJdbcClient
+{
+    private static final String IGNITE_CATALOG = "IGNITE";
+    private static final String IGNITE_SCHEMA = "PUBLIC";
+    private final AggregateFunctionRewriter aggregateFunctionRewriter;
+    private static final Splitter SPLITTER = Splitter.on(",").omitEmptyStrings().trimResults();
+
+    private final Type uuidType;
+
+    @Inject
+    public IgniteJdbcClient(
+            BaseJdbcConfig config,
+            ConnectionFactory connectionFactory,
+            IdentifierMapping identifierMapping,
+            TypeManager typeManager)
+    {
+        super(config, "`", connectionFactory, identifierMapping);
+        this.uuidType = typeManager.getType(new TypeSignature(StandardTypes.UUID));
+
+        JdbcTypeHandle bigintTypeHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
+        this.aggregateFunctionRewriter = new AggregateFunctionRewriter(
+                this::quoted,
+                ImmutableSet.<AggregateFunctionRule>builder()
+                        .add(new ImplementCountAll(bigintTypeHandle))
+                        .add(new ImplementCount(bigintTypeHandle))
+                        .add(new ImplementMinMax())
+                        .add(new ImplementSum(IgniteJdbcClient::toTypeHandle))
+                        .add(new ImplementAvgFloatingPoint())
+                        .add(new ImplementAvgDecimal())
+                        .add(new ImplementAvgBigint())
+                        .build());
+    }
+
+    @Override
+    public Collection<String> listSchemas(Connection connection)
+    {
+        return ImmutableSet.of(IGNITE_SCHEMA);
+    }
+
+    @Override
+    public ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        DatabaseMetaData metadata = connection.getMetaData();
+        return metadata.getTables(
+                IGNITE_CATALOG,
+                IGNITE_SCHEMA,
+                tableName.orElse(null),
+                new String[] {"TABLE", "VIEW"});
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+
+        switch (typeHandle.getJdbcType()) {
+            case Types.CHAR:
+                return Optional.of(charColumnMapping(typeHandle.getColumnSize()));
+            case Types.VARCHAR:
+                return Optional.of(varcharColumnMapping(typeHandle.getColumnSize()));
+            case Types.DATE:
+                return Optional.of(ColumnMapping.longMapping(
+                        DATE,
+                        dateReadFunction(),
```

Comment:
Your impl for `dateReadFunction` and `dateWriteFunction` are the same as the one in `StandardColumnMappings` (with some cases missing which can lead to incorrect results).
Why can't we use `StandardColumnMappings.dateColumnMapping` here instead?

Also, for the type mapping tests please use `SqlDataTypeTest` instead of `DataTypeTest` which is known to mask some correctness issues. You can take a look at Oracle's type mapping tests or Sql Server for examples.

Reply:
> Your impl for `dateReadFunction` and `dateWriteFunction` are the same as the one in `StandardColumnMappings` (with some cases missing which can lead to incorrect results).
> Why can't we use `StandardColumnMappings.dateColumnMapping` here instead?
> https://github.com/trinodb/trino/pull/8323#discussion_r660193163 Actually, it performance different with the original one. 

> Also, for the type mapping tests please use `SqlDataTypeTest` instead of `DataTypeTest` which is known to mask some correctness issues. You can take a look at Oracle's type mapping tests or Sql Server for examples.

Ok, will do.

Reply:
Thanks for clarifying. It seems like the driver is handling the conversion.